### ISSUE TITLE
Some WAV files fail to play (regression)

### ISSUE THREAD

**Issue by crumblingstatue** [SFML/SFML#955]
Since sometime after SFML 2.1, some WAV files that successfully played before fail to play.

Here is the code I've used to test:
```cpp
#include <SFML/Audio.hpp>
#include <iostream>

int main() {
	sf::SoundBuffer buf;
	if (!buf.loadFromFile("Startup.wav")) {
		std::cerr << "Fatal error: Could not load sound file.";
		return 1;
	}
	sf::Sound sound(buf);
	sound.play();
	sf::sleep(sf::milliseconds(2000));
}

```
The files load without error, but when playing them, only silence can be heard.

I've uploaded 2 sample WAV files that fail to play to http://www.filedropper.com/bad.

I've tried bisecting, but I came across commits that make the test code fail to build, so there are multiple candidates:

6096d220913ec2d29037ddc0cc9e97f45ab903e2 (Definitely not the one, only touches MSVC extlibs)
5e0e645f46b6ad30ef40c620b04b2e0b406b5e1c (Adds FLAC support, probably doesn't touch WAV?)
f0608eaed86b6423852121eefa4d20cd12dd3af9 (This is my suspect)
3fbfde39a5aa167f9cd606b85ad0118411831a96

Tested on Arch Linux, x86_64

**Comment by madpew** [comment]
From looking at the 2 example files given I suspect it might have something to do with either both files having a bitdepth of 24 instead of the default 16 and/or higher than usual samplingrates (48k and 96k) instead of the 44k default. But I'm just guessing.

**Comment by LaurentGomila** [comment]
Indeed, the WAV reader currently supports 8, 16 and 32 bits samples, but not 24.

**Comment by Foaly** [comment]
I implemented a fix for this. If you want to you can check it out and test it.

**Comment by Foaly** [comment]
@crumblingstatue Hello I tryed to download your files to test if my fix solves the problem, but the download link is dead. Could you reupload the files, so we can be sure. Thanks!

**Comment by crumblingstatue** [comment]
@Foaly I tested your branch, the sound now plays audibly.
I also uploaded one of the sounds here: https://hostr.co/EdbamoWFmfuf

**Comment by Foaly** [comment]
Cool! Thank you very much. Glad to hear it fixes your problem.

**Comment by eXpl0it3r** [comment]
Fixed in b7d7ac44f311165f2c1a5f2b37a21410cfab2e74
